--- tools/nikon-firmware-tools/Nikon-Patch-JS/src/nikon_patch/patches.c
+++ patchcli/patches.c
@@ -3835,13 +3835,285 @@
 uint8_t D800_0111_003_change_011_a[] = {0x02,0x31,0x2D,0x00};
 struct Change D800_0111_003_change_011 = CHANGE(1, 0x021EC0, D800_0111_003_change_011_b, D800_0111_003_change_011_a);
 struct Change const * D800_0111_003[] = {&D800_0111_003_change_000,&D800_0111_003_change_001,&D800_0111_003_change_002,&D800_0111_003_change_003,&D800_0111_003_change_004,&D800_0111_003_change_005,&D800_0111_003_change_006,&D800_0111_003_change_007,&D800_0111_003_change_008,&D800_0111_003_change_009,&D800_0111_003_change_010,&D800_0111_003_change_011};
+
+uint8_t D800_0111_004_change_000_b[] = {0x01,0x6E,0x36,0x00};
+uint8_t D800_0111_004_change_000_a[] = {0x03,0xD0,0x90,0x00};
+struct Change D800_0111_004_change_000 = CHANGE(1, 0x021F38, D800_0111_004_change_000_b, D800_0111_004_change_000_a);
+uint8_t D800_0111_004_change_001_b[] = {0x01,0x31,0x2D,0x00};
+uint8_t D800_0111_004_change_001_a[] = {0x03,0x93,0x87,0x00};
+struct Change D800_0111_004_change_001 = CHANGE(1, 0x021F3E, D800_0111_004_change_001_b, D800_0111_004_change_001_a);
+uint8_t D800_0111_004_change_002_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800_0111_004_change_002_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800_0111_004_change_002 = CHANGE(1, 0x021F4C, D800_0111_004_change_002_b, D800_0111_004_change_002_a);
+uint8_t D800_0111_004_change_003_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800_0111_004_change_003_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800_0111_004_change_003 = CHANGE(1, 0x021F52, D800_0111_004_change_003_b, D800_0111_004_change_003_a);
+uint8_t D800_0111_004_change_004_b[] = {0x01,0x6E,0x36,0x00};
+uint8_t D800_0111_004_change_004_a[] = {0x03,0xD0,0x90,0x00};
+struct Change D800_0111_004_change_004 = CHANGE(1, 0x021FB0, D800_0111_004_change_004_b, D800_0111_004_change_004_a);
+uint8_t D800_0111_004_change_005_b[] = {0x01,0x31,0x2D,0x00};
+uint8_t D800_0111_004_change_005_a[] = {0x03,0x93,0x87,0x00};
+struct Change D800_0111_004_change_005 = CHANGE(1, 0x021FB6, D800_0111_004_change_005_b, D800_0111_004_change_005_a);
+uint8_t D800_0111_004_change_006_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800_0111_004_change_006_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800_0111_004_change_006 = CHANGE(1, 0x021FC4, D800_0111_004_change_006_b, D800_0111_004_change_006_a);
+uint8_t D800_0111_004_change_007_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800_0111_004_change_007_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800_0111_004_change_007 = CHANGE(1, 0x021FCA, D800_0111_004_change_007_b, D800_0111_004_change_007_a);
+struct Change const * D800_0111_004[] = {&D800_0111_004_change_000,&D800_0111_004_change_001,&D800_0111_004_change_002,&D800_0111_004_change_003,&D800_0111_004_change_004,&D800_0111_004_change_005,&D800_0111_004_change_006,&D800_0111_004_change_007};
+
+uint8_t D800_0111_005_change_000_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800_0111_005_change_000_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800_0111_005_change_000 = CHANGE(1, 0x021F64, D800_0111_005_change_000_b, D800_0111_005_change_000_a);
+uint8_t D800_0111_005_change_001_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800_0111_005_change_001_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800_0111_005_change_001 = CHANGE(1, 0x021F6A, D800_0111_005_change_001_b, D800_0111_005_change_001_a);
+uint8_t D800_0111_005_change_002_b[] = {0x00,0x7A,0x12,0x00};
+uint8_t D800_0111_005_change_002_a[] = {0x00,0xB7,0x1B,0x00};
+struct Change D800_0111_005_change_002 = CHANGE(1, 0x021F78, D800_0111_005_change_002_b, D800_0111_005_change_002_a);
+uint8_t D800_0111_005_change_003_b[] = {0x00,0x5B,0x8D,0x80};
+uint8_t D800_0111_005_change_003_a[] = {0x00,0x98,0x96,0x80};
+struct Change D800_0111_005_change_003 = CHANGE(1, 0x021F7E, D800_0111_005_change_003_b, D800_0111_005_change_003_a);
+uint8_t D800_0111_005_change_004_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800_0111_005_change_004_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800_0111_005_change_004 = CHANGE(1, 0x021FDC, D800_0111_005_change_004_b, D800_0111_005_change_004_a);
+uint8_t D800_0111_005_change_005_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800_0111_005_change_005_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800_0111_005_change_005 = CHANGE(1, 0x021FE2, D800_0111_005_change_005_b, D800_0111_005_change_005_a);
+uint8_t D800_0111_005_change_006_b[] = {0x00,0x7A,0x12,0x00};
+uint8_t D800_0111_005_change_006_a[] = {0x00,0xB7,0x1B,0x00};
+struct Change D800_0111_005_change_006 = CHANGE(1, 0x021FF0, D800_0111_005_change_006_b, D800_0111_005_change_006_a);
+uint8_t D800_0111_005_change_007_b[] = {0x00,0x5B,0x8D,0x80};
+uint8_t D800_0111_005_change_007_a[] = {0x00,0x98,0x96,0x80};
+struct Change D800_0111_005_change_007 = CHANGE(1, 0x021FF6, D800_0111_005_change_007_b, D800_0111_005_change_007_a);
+struct Change const * D800_0111_005[] = {&D800_0111_005_change_000,&D800_0111_005_change_001,&D800_0111_005_change_002,&D800_0111_005_change_003,&D800_0111_005_change_004,&D800_0111_005_change_005,&D800_0111_005_change_006,&D800_0111_005_change_007};
 
+
 struct Patch D800_0111_patches[] = {
     {.id = 1, .level = Alpha, .name="Video 1080 HQ 36mbps Bit-rate NQ old HQ", .blocks={2,3,4}, .changes=D800_0111_000, .changes_len=(sizeof(D800_0111_000)/sizeof(struct Change*))}
     ,{.id = 2, .level = Alpha, .name="Video 1080 HQ 54mbps Bit-rate NQ old HQ", .blocks={1,3,4}, .changes=D800_0111_001, .changes_len=(sizeof(D800_0111_001)/sizeof(struct Change*))}
     ,{.id = 3, .level = Alpha, .name="Video 1080 HQ 64mbps Bit-rate NQ old HQ", .blocks={1,2,4}, .changes=D800_0111_002, .changes_len=(sizeof(D800_0111_002)/sizeof(struct Change*))}
     ,{.id = 4, .level = Alpha, .name="Video 1080 HQ 64mbps, NQ 36mbps", .blocks={3,1,2}, .changes=D800_0111_003, .changes_len=(sizeof(D800_0111_003)/sizeof(struct Change*))}
+    ,{.id = 5, .level = Alpha, .name="Video 720p60/50 HQ 64mbps Bit-rate NQ old HQ", .blocks={1,2,4}, .changes=D800_0111_004, .changes_len=(sizeof(D800_0111_004)/sizeof(struct Change*))}
+    ,{.id = 6, .level = Alpha, .name="Video 720p30/25 HQ 24mbps Bit-rate NQ old HQ", .blocks={1,2,4}, .changes=D800_0111_005, .changes_len=(sizeof(D800_0111_005)/sizeof(struct Change*))}
+};
+
+
+uint8_t D800E_0111_000_change_000_b[] = {0x01,0x6E,0x36,0x00};
+uint8_t D800E_0111_000_change_000_a[] = {0x02,0x6E,0x36,0x00};
+struct Change D800E_0111_000_change_000 = CHANGE(1, 0x021E2E, D800E_0111_000_change_000_b, D800E_0111_000_change_000_a);
+uint8_t D800E_0111_000_change_001_b[] = {0x01,0x31,0x2D,0x00};
+uint8_t D800E_0111_000_change_001_a[] = {0x02,0x31,0x2D,0x00};
+struct Change D800E_0111_000_change_001 = CHANGE(1, 0x021E34, D800E_0111_000_change_001_b, D800E_0111_000_change_001_a);
+uint8_t D800E_0111_000_change_002_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0111_000_change_002_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800E_0111_000_change_002 = CHANGE(1, 0x021E42, D800E_0111_000_change_002_b, D800E_0111_000_change_002_a);
+uint8_t D800E_0111_000_change_003_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0111_000_change_003_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800E_0111_000_change_003 = CHANGE(1, 0x021E48, D800E_0111_000_change_003_b, D800E_0111_000_change_003_a);
+uint8_t D800E_0111_000_change_004_b[] = {0x01,0x6E,0x36,0x00};
+uint8_t D800E_0111_000_change_004_a[] = {0x02,0x6E,0x36,0x00};
+struct Change D800E_0111_000_change_004 = CHANGE(1, 0x021E5A, D800E_0111_000_change_004_b, D800E_0111_000_change_004_a);
+uint8_t D800E_0111_000_change_005_b[] = {0x01,0x31,0x2D,0x00};
+uint8_t D800E_0111_000_change_005_a[] = {0x02,0x31,0x2D,0x00};
+struct Change D800E_0111_000_change_005 = CHANGE(1, 0x021E60, D800E_0111_000_change_005_b, D800E_0111_000_change_005_a);
+uint8_t D800E_0111_000_change_006_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0111_000_change_006_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800E_0111_000_change_006 = CHANGE(1, 0x021E6E, D800E_0111_000_change_006_b, D800E_0111_000_change_006_a);
+uint8_t D800E_0111_000_change_007_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0111_000_change_007_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800E_0111_000_change_007 = CHANGE(1, 0x021E74, D800E_0111_000_change_007_b, D800E_0111_000_change_007_a);
+uint8_t D800E_0111_000_change_008_b[] = {0x01,0x6E,0x36,0x00};
+uint8_t D800E_0111_000_change_008_a[] = {0x02,0x6E,0x36,0x00};
+struct Change D800E_0111_000_change_008 = CHANGE(1, 0x021EA6, D800E_0111_000_change_008_b, D800E_0111_000_change_008_a);
+uint8_t D800E_0111_000_change_009_b[] = {0x01,0x31,0x2D,0x00};
+uint8_t D800E_0111_000_change_009_a[] = {0x02,0x31,0x2D,0x00};
+struct Change D800E_0111_000_change_009 = CHANGE(1, 0x021EAC, D800E_0111_000_change_009_b, D800E_0111_000_change_009_a);
+uint8_t D800E_0111_000_change_010_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0111_000_change_010_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800E_0111_000_change_010 = CHANGE(1, 0x021EBA, D800E_0111_000_change_010_b, D800E_0111_000_change_010_a);
+uint8_t D800E_0111_000_change_011_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0111_000_change_011_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800E_0111_000_change_011 = CHANGE(1, 0x021EC0, D800E_0111_000_change_011_b, D800E_0111_000_change_011_a);
+struct Change const * D800E_0111_000[] = {&D800E_0111_000_change_000,&D800E_0111_000_change_001,&D800E_0111_000_change_002,&D800E_0111_000_change_003,&D800E_0111_000_change_004,&D800E_0111_000_change_005,&D800E_0111_000_change_006,&D800E_0111_000_change_007,&D800E_0111_000_change_008,&D800E_0111_000_change_009,&D800E_0111_000_change_010,&D800E_0111_000_change_011};
+
+uint8_t D800E_0111_001_change_000_b[] = {0x01,0x6E,0x36,0x00};
+uint8_t D800E_0111_001_change_000_a[] = {0x03,0x6E,0x36,0x00};
+struct Change D800E_0111_001_change_000 = CHANGE(1, 0x021E2E, D800E_0111_001_change_000_b, D800E_0111_001_change_000_a);
+uint8_t D800E_0111_001_change_001_b[] = {0x01,0x31,0x2D,0x00};
+uint8_t D800E_0111_001_change_001_a[] = {0x03,0x31,0x2D,0x00};
+struct Change D800E_0111_001_change_001 = CHANGE(1, 0x021E34, D800E_0111_001_change_001_b, D800E_0111_001_change_001_a);
+uint8_t D800E_0111_001_change_002_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0111_001_change_002_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800E_0111_001_change_002 = CHANGE(1, 0x021E42, D800E_0111_001_change_002_b, D800E_0111_001_change_002_a);
+uint8_t D800E_0111_001_change_003_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0111_001_change_003_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800E_0111_001_change_003 = CHANGE(1, 0x021E48, D800E_0111_001_change_003_b, D800E_0111_001_change_003_a);
+uint8_t D800E_0111_001_change_004_b[] = {0x01,0x6E,0x36,0x00};
+uint8_t D800E_0111_001_change_004_a[] = {0x03,0x6E,0x36,0x00};
+struct Change D800E_0111_001_change_004 = CHANGE(1, 0x021E5A, D800E_0111_001_change_004_b, D800E_0111_001_change_004_a);
+uint8_t D800E_0111_001_change_005_b[] = {0x01,0x31,0x2D,0x00};
+uint8_t D800E_0111_001_change_005_a[] = {0x03,0x31,0x2D,0x00};
+struct Change D800E_0111_001_change_005 = CHANGE(1, 0x021E60, D800E_0111_001_change_005_b, D800E_0111_001_change_005_a);
+uint8_t D800E_0111_001_change_006_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0111_001_change_006_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800E_0111_001_change_006 = CHANGE(1, 0x021E6E, D800E_0111_001_change_006_b, D800E_0111_001_change_006_a);
+uint8_t D800E_0111_001_change_007_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0111_001_change_007_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800E_0111_001_change_007 = CHANGE(1, 0x021E74, D800E_0111_001_change_007_b, D800E_0111_001_change_007_a);
+uint8_t D800E_0111_001_change_008_b[] = {0x01,0x6E,0x36,0x00};
+uint8_t D800E_0111_001_change_008_a[] = {0x03,0x6E,0x36,0x00};
+struct Change D800E_0111_001_change_008 = CHANGE(1, 0x021EA6, D800E_0111_001_change_008_b, D800E_0111_001_change_008_a);
+uint8_t D800E_0111_001_change_009_b[] = {0x01,0x31,0x2D,0x00};
+uint8_t D800E_0111_001_change_009_a[] = {0x03,0x31,0x2D,0x00};
+struct Change D800E_0111_001_change_009 = CHANGE(1, 0x021EAC, D800E_0111_001_change_009_b, D800E_0111_001_change_009_a);
+uint8_t D800E_0111_001_change_010_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0111_001_change_010_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800E_0111_001_change_010 = CHANGE(1, 0x021EBA, D800E_0111_001_change_010_b, D800E_0111_001_change_010_a);
+uint8_t D800E_0111_001_change_011_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0111_001_change_011_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800E_0111_001_change_011 = CHANGE(1, 0x021EC0, D800E_0111_001_change_011_b, D800E_0111_001_change_011_a);
+struct Change const * D800E_0111_001[] = {&D800E_0111_001_change_000,&D800E_0111_001_change_001,&D800E_0111_001_change_002,&D800E_0111_001_change_003,&D800E_0111_001_change_004,&D800E_0111_001_change_005,&D800E_0111_001_change_006,&D800E_0111_001_change_007,&D800E_0111_001_change_008,&D800E_0111_001_change_009,&D800E_0111_001_change_010,&D800E_0111_001_change_011};
+
+uint8_t D800E_0111_002_change_000_b[] = {0x01,0x6E,0x36,0x00};
+uint8_t D800E_0111_002_change_000_a[] = {0x03,0xD0,0x90,0x00};
+struct Change D800E_0111_002_change_000 = CHANGE(1, 0x021E2E, D800E_0111_002_change_000_b, D800E_0111_002_change_000_a);
+uint8_t D800E_0111_002_change_001_b[] = {0x01,0x31,0x2D,0x00};
+uint8_t D800E_0111_002_change_001_a[] = {0x03,0x93,0x87,0x00};
+struct Change D800E_0111_002_change_001 = CHANGE(1, 0x021E34, D800E_0111_002_change_001_b, D800E_0111_002_change_001_a);
+uint8_t D800E_0111_002_change_002_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0111_002_change_002_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800E_0111_002_change_002 = CHANGE(1, 0x021E42, D800E_0111_002_change_002_b, D800E_0111_002_change_002_a);
+uint8_t D800E_0111_002_change_003_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0111_002_change_003_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800E_0111_002_change_003 = CHANGE(1, 0x021E48, D800E_0111_002_change_003_b, D800E_0111_002_change_003_a);
+uint8_t D800E_0111_002_change_004_b[] = {0x01,0x6E,0x36,0x00};
+uint8_t D800E_0111_002_change_004_a[] = {0x03,0xD0,0x90,0x00};
+struct Change D800E_0111_002_change_004 = CHANGE(1, 0x021E5A, D800E_0111_002_change_004_b, D800E_0111_002_change_004_a);
+uint8_t D800E_0111_002_change_005_b[] = {0x01,0x31,0x2D,0x00};
+uint8_t D800E_0111_002_change_005_a[] = {0x03,0x93,0x87,0x00};
+struct Change D800E_0111_002_change_005 = CHANGE(1, 0x021E60, D800E_0111_002_change_005_b, D800E_0111_002_change_005_a);
+uint8_t D800E_0111_002_change_006_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0111_002_change_006_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800E_0111_002_change_006 = CHANGE(1, 0x021E6E, D800E_0111_002_change_006_b, D800E_0111_002_change_006_a);
+uint8_t D800E_0111_002_change_007_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0111_002_change_007_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800E_0111_002_change_007 = CHANGE(1, 0x021E74, D800E_0111_002_change_007_b, D800E_0111_002_change_007_a);
+uint8_t D800E_0111_002_change_008_b[] = {0x01,0x6E,0x36,0x00};
+uint8_t D800E_0111_002_change_008_a[] = {0x03,0xD0,0x90,0x00};
+struct Change D800E_0111_002_change_008 = CHANGE(1, 0x021EA6, D800E_0111_002_change_008_b, D800E_0111_002_change_008_a);
+uint8_t D800E_0111_002_change_009_b[] = {0x01,0x31,0x2D,0x00};
+uint8_t D800E_0111_002_change_009_a[] = {0x03,0x93,0x87,0x00};
+struct Change D800E_0111_002_change_009 = CHANGE(1, 0x021EAC, D800E_0111_002_change_009_b, D800E_0111_002_change_009_a);
+uint8_t D800E_0111_002_change_010_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0111_002_change_010_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800E_0111_002_change_010 = CHANGE(1, 0x021EBA, D800E_0111_002_change_010_b, D800E_0111_002_change_010_a);
+uint8_t D800E_0111_002_change_011_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0111_002_change_011_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800E_0111_002_change_011 = CHANGE(1, 0x021EC0, D800E_0111_002_change_011_b, D800E_0111_002_change_011_a);
+struct Change const * D800E_0111_002[] = {&D800E_0111_002_change_000,&D800E_0111_002_change_001,&D800E_0111_002_change_002,&D800E_0111_002_change_003,&D800E_0111_002_change_004,&D800E_0111_002_change_005,&D800E_0111_002_change_006,&D800E_0111_002_change_007,&D800E_0111_002_change_008,&D800E_0111_002_change_009,&D800E_0111_002_change_010,&D800E_0111_002_change_011};
+
+uint8_t D800E_0111_003_change_000_b[] = {0x01,0x6E,0x36,0x00};
+uint8_t D800E_0111_003_change_000_a[] = {0x03,0xD0,0x90,0x00};
+struct Change D800E_0111_003_change_000 = CHANGE(1, 0x021E2E, D800E_0111_003_change_000_b, D800E_0111_003_change_000_a);
+uint8_t D800E_0111_003_change_001_b[] = {0x01,0x31,0x2D,0x00};
+uint8_t D800E_0111_003_change_001_a[] = {0x03,0x93,0x87,0x00};
+struct Change D800E_0111_003_change_001 = CHANGE(1, 0x021E34, D800E_0111_003_change_001_b, D800E_0111_003_change_001_a);
+uint8_t D800E_0111_003_change_002_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0111_003_change_002_a[] = {0x02,0x6E,0x36,0x00};
+struct Change D800E_0111_003_change_002 = CHANGE(1, 0x021E42, D800E_0111_003_change_002_b, D800E_0111_003_change_002_a);
+uint8_t D800E_0111_003_change_003_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0111_003_change_003_a[] = {0x02,0x31,0x2D,0x00};
+struct Change D800E_0111_003_change_003 = CHANGE(1, 0x021E48, D800E_0111_003_change_003_b, D800E_0111_003_change_003_a);
+uint8_t D800E_0111_003_change_004_b[] = {0x01,0x6E,0x36,0x00};
+uint8_t D800E_0111_003_change_004_a[] = {0x03,0xD0,0x90,0x00};
+struct Change D800E_0111_003_change_004 = CHANGE(1, 0x021E5A, D800E_0111_003_change_004_b, D800E_0111_003_change_004_a);
+uint8_t D800E_0111_003_change_005_b[] = {0x01,0x31,0x2D,0x00};
+uint8_t D800E_0111_003_change_005_a[] = {0x03,0x93,0x87,0x00};
+struct Change D800E_0111_003_change_005 = CHANGE(1, 0x021E60, D800E_0111_003_change_005_b, D800E_0111_003_change_005_a);
+uint8_t D800E_0111_003_change_006_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0111_003_change_006_a[] = {0x02,0x6E,0x36,0x00};
+struct Change D800E_0111_003_change_006 = CHANGE(1, 0x021E6E, D800E_0111_003_change_006_b, D800E_0111_003_change_006_a);
+uint8_t D800E_0111_003_change_007_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0111_003_change_007_a[] = {0x02,0x31,0x2D,0x00};
+struct Change D800E_0111_003_change_007 = CHANGE(1, 0x021E74, D800E_0111_003_change_007_b, D800E_0111_003_change_007_a);
+uint8_t D800E_0111_003_change_008_b[] = {0x01,0x6E,0x36,0x00};
+uint8_t D800E_0111_003_change_008_a[] = {0x03,0xD0,0x90,0x00};
+struct Change D800E_0111_003_change_008 = CHANGE(1, 0x021EA6, D800E_0111_003_change_008_b, D800E_0111_003_change_008_a);
+uint8_t D800E_0111_003_change_009_b[] = {0x01,0x31,0x2D,0x00};
+uint8_t D800E_0111_003_change_009_a[] = {0x03,0x93,0x87,0x00};
+struct Change D800E_0111_003_change_009 = CHANGE(1, 0x021EAC, D800E_0111_003_change_009_b, D800E_0111_003_change_009_a);
+uint8_t D800E_0111_003_change_010_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0111_003_change_010_a[] = {0x02,0x6E,0x36,0x00};
+struct Change D800E_0111_003_change_010 = CHANGE(1, 0x021EBA, D800E_0111_003_change_010_b, D800E_0111_003_change_010_a);
+uint8_t D800E_0111_003_change_011_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0111_003_change_011_a[] = {0x02,0x31,0x2D,0x00};
+struct Change D800E_0111_003_change_011 = CHANGE(1, 0x021EC0, D800E_0111_003_change_011_b, D800E_0111_003_change_011_a);
+struct Change const * D800E_0111_003[] = {&D800E_0111_003_change_000,&D800E_0111_003_change_001,&D800E_0111_003_change_002,&D800E_0111_003_change_003,&D800E_0111_003_change_004,&D800E_0111_003_change_005,&D800E_0111_003_change_006,&D800E_0111_003_change_007,&D800E_0111_003_change_008,&D800E_0111_003_change_009,&D800E_0111_003_change_010,&D800E_0111_003_change_011};
+
+uint8_t D800E_0111_004_change_000_b[] = {0x01,0x6E,0x36,0x00};
+uint8_t D800E_0111_004_change_000_a[] = {0x03,0xD0,0x90,0x00};
+struct Change D800E_0111_004_change_000 = CHANGE(1, 0x021F38, D800E_0111_004_change_000_b, D800E_0111_004_change_000_a);
+uint8_t D800E_0111_004_change_001_b[] = {0x01,0x31,0x2D,0x00};
+uint8_t D800E_0111_004_change_001_a[] = {0x03,0x93,0x87,0x00};
+struct Change D800E_0111_004_change_001 = CHANGE(1, 0x021F3E, D800E_0111_004_change_001_b, D800E_0111_004_change_001_a);
+uint8_t D800E_0111_004_change_002_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0111_004_change_002_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800E_0111_004_change_002 = CHANGE(1, 0x021F4C, D800E_0111_004_change_002_b, D800E_0111_004_change_002_a);
+uint8_t D800E_0111_004_change_003_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0111_004_change_003_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800E_0111_004_change_003 = CHANGE(1, 0x021F52, D800E_0111_004_change_003_b, D800E_0111_004_change_003_a);
+uint8_t D800E_0111_004_change_004_b[] = {0x01,0x6E,0x36,0x00};
+uint8_t D800E_0111_004_change_004_a[] = {0x03,0xD0,0x90,0x00};
+struct Change D800E_0111_004_change_004 = CHANGE(1, 0x021FB0, D800E_0111_004_change_004_b, D800E_0111_004_change_004_a);
+uint8_t D800E_0111_004_change_005_b[] = {0x01,0x31,0x2D,0x00};
+uint8_t D800E_0111_004_change_005_a[] = {0x03,0x93,0x87,0x00};
+struct Change D800E_0111_004_change_005 = CHANGE(1, 0x021FB6, D800E_0111_004_change_005_b, D800E_0111_004_change_005_a);
+uint8_t D800E_0111_004_change_006_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0111_004_change_006_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800E_0111_004_change_006 = CHANGE(1, 0x021FC4, D800E_0111_004_change_006_b, D800E_0111_004_change_006_a);
+uint8_t D800E_0111_004_change_007_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0111_004_change_007_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800E_0111_004_change_007 = CHANGE(1, 0x021FCA, D800E_0111_004_change_007_b, D800E_0111_004_change_007_a);
+struct Change const * D800E_0111_004[] = {&D800E_0111_004_change_000,&D800E_0111_004_change_001,&D800E_0111_004_change_002,&D800E_0111_004_change_003,&D800E_0111_004_change_004,&D800E_0111_004_change_005,&D800E_0111_004_change_006,&D800E_0111_004_change_007};
+
+uint8_t D800E_0111_005_change_000_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0111_005_change_000_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800E_0111_005_change_000 = CHANGE(1, 0x021F64, D800E_0111_005_change_000_b, D800E_0111_005_change_000_a);
+uint8_t D800E_0111_005_change_001_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0111_005_change_001_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800E_0111_005_change_001 = CHANGE(1, 0x021F6A, D800E_0111_005_change_001_b, D800E_0111_005_change_001_a);
+uint8_t D800E_0111_005_change_002_b[] = {0x00,0x7A,0x12,0x00};
+uint8_t D800E_0111_005_change_002_a[] = {0x00,0xB7,0x1B,0x00};
+struct Change D800E_0111_005_change_002 = CHANGE(1, 0x021F78, D800E_0111_005_change_002_b, D800E_0111_005_change_002_a);
+uint8_t D800E_0111_005_change_003_b[] = {0x00,0x5B,0x8D,0x80};
+uint8_t D800E_0111_005_change_003_a[] = {0x00,0x98,0x96,0x80};
+struct Change D800E_0111_005_change_003 = CHANGE(1, 0x021F7E, D800E_0111_005_change_003_b, D800E_0111_005_change_003_a);
+uint8_t D800E_0111_005_change_004_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0111_005_change_004_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800E_0111_005_change_004 = CHANGE(1, 0x021FDC, D800E_0111_005_change_004_b, D800E_0111_005_change_004_a);
+uint8_t D800E_0111_005_change_005_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0111_005_change_005_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800E_0111_005_change_005 = CHANGE(1, 0x021FE2, D800E_0111_005_change_005_b, D800E_0111_005_change_005_a);
+uint8_t D800E_0111_005_change_006_b[] = {0x00,0x7A,0x12,0x00};
+uint8_t D800E_0111_005_change_006_a[] = {0x00,0xB7,0x1B,0x00};
+struct Change D800E_0111_005_change_006 = CHANGE(1, 0x021FF0, D800E_0111_005_change_006_b, D800E_0111_005_change_006_a);
+uint8_t D800E_0111_005_change_007_b[] = {0x00,0x5B,0x8D,0x80};
+uint8_t D800E_0111_005_change_007_a[] = {0x00,0x98,0x96,0x80};
+struct Change D800E_0111_005_change_007 = CHANGE(1, 0x021FF6, D800E_0111_005_change_007_b, D800E_0111_005_change_007_a);
+struct Change const * D800E_0111_005[] = {&D800E_0111_005_change_000,&D800E_0111_005_change_001,&D800E_0111_005_change_002,&D800E_0111_005_change_003,&D800E_0111_005_change_004,&D800E_0111_005_change_005,&D800E_0111_005_change_006,&D800E_0111_005_change_007};
+
+
+struct Patch D800E_0111_patches[] = {
+    {.id = 1, .level = Alpha, .name="Video 1080 HQ 36mbps Bit-rate NQ old HQ", .blocks={2,3,4}, .changes=D800E_0111_000, .changes_len=(sizeof(D800E_0111_000)/sizeof(struct Change*))}
+    ,{.id = 2, .level = Alpha, .name="Video 1080 HQ 54mbps Bit-rate NQ old HQ", .blocks={1,3,4}, .changes=D800E_0111_001, .changes_len=(sizeof(D800E_0111_001)/sizeof(struct Change*))}
+    ,{.id = 3, .level = Alpha, .name="Video 1080 HQ 64mbps Bit-rate NQ old HQ", .blocks={1,2,4}, .changes=D800E_0111_002, .changes_len=(sizeof(D800E_0111_002)/sizeof(struct Change*))}
+    ,{.id = 4, .level = Alpha, .name="Video 1080 HQ 64mbps, NQ 36mbps", .blocks={3,1,2}, .changes=D800E_0111_003, .changes_len=(sizeof(D800E_0111_003)/sizeof(struct Change*))}
+    ,{.id = 5, .level = Alpha, .name="Video 720p60/50 HQ 64mbps Bit-rate NQ old HQ", .blocks={1,2,4}, .changes=D800E_0111_004, .changes_len=(sizeof(D800E_0111_004)/sizeof(struct Change*))}
+    ,{.id = 6, .level = Alpha, .name="Video 720p30/25 HQ 24mbps Bit-rate NQ old HQ", .blocks={1,2,4}, .changes=D800E_0111_005, .changes_len=(sizeof(D800E_0111_005)/sizeof(struct Change*))}
 };
+
+struct PatchSet D800E_0111_ps = PATCHSET("D800E", "1.11", D800E_0111_patches, 0);
 
 struct Patch D800E_0101_patches[] = {
 };
@@ -4112,11 +4384,66 @@
 struct Change D800E_0110_003_change_009 = CHANGE(1, 0xB08452, D800E_0110_003_change_009_b, D800E_0110_003_change_009_a);
 struct Change const * D800E_0110_003[] = {&D800E_0110_003_change_000,&D800E_0110_003_change_001,&D800E_0110_003_change_002,&D800E_0110_003_change_003,&D800E_0110_003_change_004,&D800E_0110_003_change_005,&D800E_0110_003_change_006,&D800E_0110_003_change_007,&D800E_0110_003_change_008,&D800E_0110_003_change_009};
 
+uint8_t D800E_0110_004_change_000_b[] = {0x01,0x6E,0x36,0x00};
+uint8_t D800E_0110_004_change_000_a[] = {0x03,0xD0,0x90,0x00};
+struct Change D800E_0110_004_change_000 = CHANGE(1, 0x021F38, D800E_0110_004_change_000_b, D800E_0110_004_change_000_a);
+uint8_t D800E_0110_004_change_001_b[] = {0x01,0x31,0x2D,0x00};
+uint8_t D800E_0110_004_change_001_a[] = {0x03,0x93,0x87,0x00};
+struct Change D800E_0110_004_change_001 = CHANGE(1, 0x021F3E, D800E_0110_004_change_001_b, D800E_0110_004_change_001_a);
+uint8_t D800E_0110_004_change_002_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0110_004_change_002_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800E_0110_004_change_002 = CHANGE(1, 0x021F4C, D800E_0110_004_change_002_b, D800E_0110_004_change_002_a);
+uint8_t D800E_0110_004_change_003_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0110_004_change_003_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800E_0110_004_change_003 = CHANGE(1, 0x021F52, D800E_0110_004_change_003_b, D800E_0110_004_change_003_a);
+uint8_t D800E_0110_004_change_004_b[] = {0x01,0x6E,0x36,0x00};
+uint8_t D800E_0110_004_change_004_a[] = {0x03,0xD0,0x90,0x00};
+struct Change D800E_0110_004_change_004 = CHANGE(1, 0x021FB0, D800E_0110_004_change_004_b, D800E_0110_004_change_004_a);
+uint8_t D800E_0110_004_change_005_b[] = {0x01,0x31,0x2D,0x00};
+uint8_t D800E_0110_004_change_005_a[] = {0x03,0x93,0x87,0x00};
+struct Change D800E_0110_004_change_005 = CHANGE(1, 0x021FB6, D800E_0110_004_change_005_b, D800E_0110_004_change_005_a);
+uint8_t D800E_0110_004_change_006_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0110_004_change_006_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800E_0110_004_change_006 = CHANGE(1, 0x021FC4, D800E_0110_004_change_006_b, D800E_0110_004_change_006_a);
+uint8_t D800E_0110_004_change_007_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0110_004_change_007_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800E_0110_004_change_007 = CHANGE(1, 0x021FCA, D800E_0110_004_change_007_b, D800E_0110_004_change_007_a);
+struct Change const * D800E_0110_004[] = {&D800E_0110_004_change_000,&D800E_0110_004_change_001,&D800E_0110_004_change_002,&D800E_0110_004_change_003,&D800E_0110_004_change_004,&D800E_0110_004_change_005,&D800E_0110_004_change_006,&D800E_0110_004_change_007};
+
+uint8_t D800E_0110_005_change_000_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0110_005_change_000_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800E_0110_005_change_000 = CHANGE(1, 0x021F64, D800E_0110_005_change_000_b, D800E_0110_005_change_000_a);
+uint8_t D800E_0110_005_change_001_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0110_005_change_001_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800E_0110_005_change_001 = CHANGE(1, 0x021F6A, D800E_0110_005_change_001_b, D800E_0110_005_change_001_a);
+uint8_t D800E_0110_005_change_002_b[] = {0x00,0x7A,0x12,0x00};
+uint8_t D800E_0110_005_change_002_a[] = {0x00,0xB7,0x1B,0x00};
+struct Change D800E_0110_005_change_002 = CHANGE(1, 0x021F78, D800E_0110_005_change_002_b, D800E_0110_005_change_002_a);
+uint8_t D800E_0110_005_change_003_b[] = {0x00,0x5B,0x8D,0x80};
+uint8_t D800E_0110_005_change_003_a[] = {0x00,0x98,0x96,0x80};
+struct Change D800E_0110_005_change_003 = CHANGE(1, 0x021F7E, D800E_0110_005_change_003_b, D800E_0110_005_change_003_a);
+uint8_t D800E_0110_005_change_004_b[] = {0x00,0xB7,0x1B,0x00};
+uint8_t D800E_0110_005_change_004_a[] = {0x01,0x6E,0x36,0x00};
+struct Change D800E_0110_005_change_004 = CHANGE(1, 0x021FDC, D800E_0110_005_change_004_b, D800E_0110_005_change_004_a);
+uint8_t D800E_0110_005_change_005_b[] = {0x00,0x98,0x96,0x80};
+uint8_t D800E_0110_005_change_005_a[] = {0x01,0x31,0x2D,0x00};
+struct Change D800E_0110_005_change_005 = CHANGE(1, 0x021FE2, D800E_0110_005_change_005_b, D800E_0110_005_change_005_a);
+uint8_t D800E_0110_005_change_006_b[] = {0x00,0x7A,0x12,0x00};
+uint8_t D800E_0110_005_change_006_a[] = {0x00,0xB7,0x1B,0x00};
+struct Change D800E_0110_005_change_006 = CHANGE(1, 0x021FF0, D800E_0110_005_change_006_b, D800E_0110_005_change_006_a);
+uint8_t D800E_0110_005_change_007_b[] = {0x00,0x5B,0x8D,0x80};
+uint8_t D800E_0110_005_change_007_a[] = {0x00,0x98,0x96,0x80};
+struct Change D800E_0110_005_change_007 = CHANGE(1, 0x021FF6, D800E_0110_005_change_007_b, D800E_0110_005_change_007_a);
+struct Change const * D800E_0110_005[] = {&D800E_0110_005_change_000,&D800E_0110_005_change_001,&D800E_0110_005_change_002,&D800E_0110_005_change_003,&D800E_0110_005_change_004,&D800E_0110_005_change_005,&D800E_0110_005_change_006,&D800E_0110_005_change_007};
+
+
 struct Patch D800E_0110_patches[] = {
     {.id = 1, .level = Released, .name="Video 1080 HQ 36mbps Bit-rate NQ old HQ", .blocks={2,3}, .changes=D800E_0110_000, .changes_len=(sizeof(D800E_0110_000)/sizeof(struct Change*))}
     ,{.id = 2, .level = Released, .name="Video 1080 HQ 54mbps Bit-rate NQ old HQ", .blocks={1,3}, .changes=D800E_0110_001, .changes_len=(sizeof(D800E_0110_001)/sizeof(struct Change*))}
     ,{.id = 3, .level = Released, .name="Video 1080 HQ 64mbps Bit-rate NQ old HQ", .blocks={1,2}, .changes=D800E_0110_002, .changes_len=(sizeof(D800E_0110_002)/sizeof(struct Change*))}
     ,{.id = 4, .level = Beta, .name="True Dark Current - Menu based", .blocks={}, .changes=D800E_0110_003, .changes_len=(sizeof(D800E_0110_003)/sizeof(struct Change*))}
+    ,{.id = 5, .level = Alpha, .name="Video 720p60/50 HQ 64mbps Bit-rate NQ old HQ", .blocks={1,2,4}, .changes=D800E_0110_004, .changes_len=(sizeof(D800E_0110_004)/sizeof(struct Change*))}
+    ,{.id = 6, .level = Alpha, .name="Video 720p30/25 HQ 24mbps Bit-rate NQ old HQ", .blocks={1,2,4}, .changes=D800E_0110_005, .changes_len=(sizeof(D800E_0110_005)/sizeof(struct Change*))}
 };
 
 struct Patch D810_0102_patches[] = {
@@ -4378,5 +4705,6 @@
      ,{.id = 54, .hash = {0xBB,0xB0,0x44,0x5F,0x58,0x18,0x57,0xCD,0x0A,0xF7,0x76,0x9D,0xEF,0xBD,0x09,0x51}, .patches = &D4S_0101_ps}
      ,{.id = 55, .hash = {0x56,0xCB,0xD3,0xF0,0xBA,0x4B,0x2B,0x74,0x27,0x4F,0xC3,0xEA,0xE4,0x4F,0xD7,0x9B}, .patches = &D4S_0132_ps}
      ,{.id = 56, .hash = {0x09,0xA6,0x90,0xCE,0x56,0x06,0xF2,0x46,0xCB,0x1B,0x51,0x45,0xFD,0x27,0x6E,0xD5}, .patches = &D5_0120_ps}
+     ,{.id = 57, .hash = {0x1B,0x03,0x3E,0xB7,0x79,0x5B,0x13,0xAA,0x13,0x7A,0x91,0x70,0x03,0x2B,0xA2,0xA7}, .patches = &D800E_0111_ps}
 };
 const uint32_t patches_count = sizeof(patches)/sizeof(struct PatchMap);
